packages/typespec-client-generator-core/lib/decorators.tsp

```diff
@@ -608,3 +608,18 @@ extern dec scope(target: Operation, scope?: valueof string);
  * ```
  */
 extern dec apiVersion(target: ModelProperty, value?: valueof boolean, scope?: valueof string);
+
+/**
+ * Whether a model needs to treat empty string as null.
+ * @param scope The language scope you want this decorator to apply to. If not specified, will apply to all language emitters.
+ * You can use "!" to specify negation such as "!(java, python)" or "!java, !python".
+ *
+ * @example
+ * ```typespec
+ * @emptyStringAsNull
+ * model MyModel {
+ *   prop: string;
+ * }
+ * ```
+ */
+extern dec emptyStringAsNull(target: Model, scope?: valueof string);
```

Comment:
From feature's perspective, it makes more sense if this decorator applies to the property.
I think we made it targets the model because our old configuration is using the model name as its value, but that is actually a bad decision - because apparently, in most cases, not all the string properties in that model want to treat empty string as null.
So here I would say we should change the target to `Property`.

Reply:
As offline discussed, it does look clearer to target on property instead of model.
But autorest.csharp emitter puts this config on model now. When we need to migrate from OpenAPI to TypeSpec, we need to add the decorator on property instead of model. We can handle this during tsp conversion.

Reply:
change the decorator's target to model property.